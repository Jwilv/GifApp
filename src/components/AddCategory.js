import React, { useState } from 'react'    
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInpuntChange = element => { setInputValue(element.target.value) }

    const handleSubmit = element => {
        element.preventDefault();
        //console.log('salio bien')
        if(inputValue.trim().length > 0){
            setCategories(categorie =>[inputValue,...categorie]);
            setInputValue('')
        }
        
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                value={inputValue}
                onChange={handleInpuntChange}
            />
        </form>
    )

}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}