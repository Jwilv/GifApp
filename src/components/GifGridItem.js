import React from 'react'

export const GifGridItem = ({title, url }) => {

    console.log({url})

    return (
        <div className='card animate__animated animate__backInLeft'>
        <img src={url} alt={title}/>
        <p>{title}</p>
        </div>
    )
}
