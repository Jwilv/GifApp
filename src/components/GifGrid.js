import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h2 className='animate__animated animate__backInLeft'>{category}</h2>
      <div className='card-grid'>
        {loading && <p>loading...</p>}
        {
          images.map(img => (
            <GifGridItem
              key={img.id}
              {...img}
            />
          ))
        }
      </div>
    </>
  )
}
GifGrid.propTypes ={
  category : PropTypes.string.isRequired,
}
