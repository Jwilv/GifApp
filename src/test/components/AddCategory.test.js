import React from 'react'
import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory';

describe('test de <AddCategory />', () => {

    const setCategories = () => { }
    const wrapper = shallow(<AddCategory setCategories={setCategories} />)

    test('debe mostrarase correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    });

    test('debe de cambiar la caja de texto', () => {
        const inptu = wrapper.find('input')
        const value = 'hola mundo'
        inptu.simulate('change', { target: { value } })
        const output = wrapper.find('p').text()
        expect(value).toBe(output)

    });
});