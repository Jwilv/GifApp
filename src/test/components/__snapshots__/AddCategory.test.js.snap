// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test de <AddCategory /> debe mostrarase correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
